### PR TITLE
fix: curly offsetlookup

### LINKED ISSUE #374: invalid ast for `{` and `}`
Input:
```php
$var = $foo->bzr_{1};
$var = $foo->bzr_{'string'};
$var = $foo->bzr_{$baz};
$var = $foo->bzr_{$baz->foo};
$var = $foo->bzr_{$var ? 'one' : 'two'};
```

Parsed as encapsed with `offset`, but it is just offsetlookup with curly `brackets`

Should be parse as:
```php
$var = $foo->bzr_[1];
$var = $foo->bzr_['string'];
$var = $foo->bzr_[$baz];
$var = $foo->bzr_[$baz->foo];
$var = $foo->bzr_[$var ? 'one' : 'two'];
```

### PR DESCRIPTION
**Pull request by evilebottnawi**
It is basic tests

I find a lot of check on `[`, but array access can be done with `{`, need investigate

fixes #374

/cc @ichiriac

### CODE REVIEW

File: src/parser/variable.js [hunk 1]
```diff
@@ -209,17 +182,18 @@ module.exports = {
           }
           break;
         case "[":
+        case "{":
           node = this.node("offsetlookup");
           this.next();
           offset = false;
           if (encapsed) {
             offset = this.read_encaps_var_offset();
```
Comment (ichiriac) [inline src/parser/variable.js]: can you check `$obj{$foo]` and `$obj[$bar}` they should fail, but in this implementation they may pass

File: src/parser/variable.js [hunk 2]
```diff
@@ -209,17 +182,18 @@ module.exports = {
           }
           break;
         case "[":
+        case "{":
           node = this.node("offsetlookup");
```
Comment (ichiriac) [inline src/parser/variable.js]: can you add a new `curly` property in order to keep track of offset type of `{` or `[`

**Comment by evilebottnawi** [comment]
:+1: WIP

File: src/parser/variable.js (hunk 2 above)
Comment (evilebottnawi) [reply src/parser/variable.js]: @ichiriac we have several places with `offsetlookup`, i want to refactor them and test before implement `curly` option (maybe some of them invalid), so let's do it in other PR

Comment (ichiriac) [reply src/parser/variable.js]: ok, lets do it later - it was referenced to #380

**Pull request merged**